There doesn't seem to be a valid reason for this restriction, and it doesn't
impede building anyway.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -145,7 +145,7 @@
     "numpy==1.21.1; python_version=='3.8' and platform_python_implementation != 'PyPy'",
     "numpy==1.21.1; python_version=='3.9' and platform_python_implementation != 'PyPy'",
     "numpy==1.21.6; python_version=='3.10' and platform_system != 'Windows' and platform_python_implementation != 'PyPy'",
-    "numpy==1.23.3; python_version=='3.11' and platform_python_implementation != 'PyPy'",
+    "numpy>=1.23.3; python_version=='3.11' and platform_python_implementation != 'PyPy'",
     "numpy; python_version>='3.12'",
     "numpy; python_version>='3.8' and platform_python_implementation=='PyPy'",
 ]
